.github/workflows/distcheck.yaml

```diff
@@ -13,6 +13,11 @@ concurrency:
 
 jobs:
   amazonlinux:
+    env:
+      # note, related to issue around actions/checkout@v4, linked below. This
+      # environment variable is also now needed, as of july 2024.
+      # ref: https://github.com/actions/runner/issues/2906#issuecomment-2208546951
+      ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION: 'true'
```

Comment:
Can we restrict this to al2 (and not al2023)?

Reply:
This is blocking PRs, so prefer to leave it as-is for now and we can follow up with a more granular fix in the future.

There are no tokens/keys referenced in any of our actions (and on top of that, CI runs are allowlisted to only run for contributors), so despite the alarming name of this environment variable I don't think there's actually a security concern here for our purposes. (famous last words.)